modules/packages/Socket.chpl

```diff
@@ -242,9 +241,24 @@ proc ipAddr.writeThis(f) throws {
 }
 
 pragma "no doc"
-operator timeval.=(other: real) {
-  this.tv_sec = other:c_long;
-  this.tv_usec = (other - this.tv_sec) * 1000000;
+operator struct_timeval.=(other: real) {
+  this.tv_sec = (other:c_long):time_t;
+  this.tv_usec = ((other - this.tv_sec) * 1000000):suseconds_t;
+}
```

Comment:
I realize you're updating existing code to keep it working, so haven't added anything, but...  It seems surprising to me that this overload is a good idea since it doesn't seem to be implementing an assignment, but some sort of difference?  Put another way, any time I assigned two `timeval`s to each other, I'd expect the rhs to just copy over without any re-interpretation of its values.  I'm also not seeing obvious ways in which this module is relying on this specific overload (and I think we've even talked about making it illegal to override a compiler-generated default `=` for a record at times, though probably haven't implemented it?).  Not necessary to do here, and not related to this PR, but my instinct would be to remove this while here to see whether things still work the same as before.

Tagging @mppf as the original reviewer of the Socket module to see whether he knows something I'm not picking up on.

Reply:
I don't think we were going to make it illegal for records, are you maybe thinking of how it's illegal to overload `=` for classes?

Reply:
(maybe we relied on it more before you introduced the `indefiniteTimeout` helper?)

Reply:
> I don't think we were going to make it illegal for records, are you maybe thinking of how it's illegal to overload = for classes?

No, I'm thinking of discussions in which we were discussing tertiary methods and operators and whether the presence/absence of a compiler-generated version of an operator would prevent a user from writing their own tertiary version (to avoid confusion).

But maybe I shouldn't have brought that up at all since it's mostly an aside to the main point which was:  Why is this overload reasonable, and why would it need to be defined here rather than where `struct_timeval` was defined itself?

Reply:
I tried dropping a compilerError() into there and running the `test/library/packages/Socket/` tests, but got other compiler errors like AF_INET* being undefined (looks like it's defined in `Sys`, along with others.  @jeremiah-corrado, what testing have you done on the PR (and where?  seems like that directory is skipped in many configurations).

Reply:
I've just done standard paratesting on chapcs. I reran yesterday after the changes I made, but I still need to check those results (currently dealing with VPN issues...)

Reply:
> Why is this overload reasonable, and why would it need to be defined here rather than where `struct_timeval` was defined itself?

Oh, I'd missed that it was defined in a different location.  Yeah, I think it would be better to put the overload where `struct_timeval` is defined

Reply:
>  It seems surprising to me that this overload is a good idea since it doesn't seem to be implementing an assignment, but some sort of difference?

I think it is implementing assignment. Maybe these comments will help:

``` chapel
operator struct_timeval.=(other: real) {
  this.tv_sec = (other:c_long):time_t; // set the whole number of seconds
  this.tv_usec = ((other - this.tv_sec) * 1000000):suseconds_t; // set any fractional portion of a second
}
```

I think it would be reasonable for this to be included wherever `struct_timeval` is defined.

Reply:
Ok thanks, I'll go ahead and move it to `OS.POSIX`

Reply:
Oh, and one more thought in this chain:  I'm still not seeing any sign that this overload is actually used by Sockets, which makes me think it should just be removed rather than implying we should support for an unused/unrequested capability.  That said, I'm only basing that on inspection.

Reply:
I'm not sure if this helps in making a decision here, but it looks like `time_t` can either be defined as a `long` or `float`:(https://www.gnu.org/software/libc/manual/html_node/Time-Types.html)

So maybe this overload was intended to make it easier for users to define a `struct_timeval` using a float, then pass to one of the Socket functions?

Reply:
> So maybe this overload was intended to make it easier for users to define a `struct_timeval` using a float, then pass to one of the Socket functions?

AFAIK it was just intended as a convenience for people using `struct_timeval` from Chapel.